### DISCUSSION TITLE
How to get possible type when using fragment.

### DISCUSSION THREAD

**Discussion by raja-OX** [0no-co/gql.tada#84] [Q&A] [answered]
Hi i am getting simple example pokemon but i types i am getting is 
```ts 
[$tada.fragmentRefs]: {
            PokemonItem: unique symbol;
};
```

May i know how to setup this correctly? I am using apollo client. 

![Screenshot from 2024-02-27 12-19-29](https://github.com/0no-co/gql.tada/assets/112232124/7f045854-8672-4eca-9436-ea280ab98a6b)
![Screenshot from 2024-02-27 12-19-42](https://github.com/0no-co/gql.tada/assets/112232124/9de9b0c5-fb02-4388-9293-699072c3b0ae)
![Screenshot from 2024-02-27 12-20-05](https://github.com/0no-co/gql.tada/assets/112232124/38fbbb4a-ef9a-4fbc-b220-788dc499c2ab)

Sample code
```js
const PokemonItemFragment = graphql(`
  fragment PokemonItem on Pokemon {
    id
    name
  }
`);

const PokemonsQuery = graphql(`
  query Pokemons ($limit: Int = 10) {
    pokemons(limit: $limit) {
      ...PokemonItem
    }
  }
`, [PokemonItemFragment]);

const Home = () => {
	const { data } = useQuery(PokemonsQuery, {
	
	});
	return (
	<div>
	   {data?.pokemons?.map((pokemon)=>{
                  return pokemon?.
             })}
	</div>
	);
};

export default Home;
```

**Comment by JoviDeCroock** [comment, marked as answer]
This is detailed in the documentation, we opt you into [`fragment co-location`](https://gql-tada.0no.co/guides/fragment-colocation/) by default. You can opt out of this with the `@_unmask` directive (which is also on that docs page).

```js
const PokemonItemFragment = graphql(`
  fragment PokemonItem on Pokemon @_unmask {
    id
    name
  }
`);
```

The symbols thing is so that you'd use `readFragment` in your accompanying component and that it would insist you pass in the correct piece of data to said component. This is a style of GraphQL which is popular in the Relay ecosystem and helps us define the data requirements for reusable components. The guild has a very good blog post about this over [here](https://the-guild.dev/blog/unleash-the-power-of-fragments-with-graphql-codegen).
